qa/tasks/rgw_reshard.py

```diff
@@ -0,0 +1,328 @@
+    log.debug(' test: reshard add')
+    old_user_stats = get_user_stats(ctx, client, user)
+    num_shards = bucket_stats1.num_shards + 1
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'add', '--bucket', bucket_name1, '--num-shards', str(num_shards)], check_status=True)
+
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'list'], check_status=True)
+    assert len(out) == 1
+    assert out[0]['bucket_name'] == bucket_name1
+
+    # TESTCASE 'reshard-process','reshard','','process bucket resharding','succeeds'
+    log.debug(' test: reshard process')
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'process', '--bucket', bucket_name1], check_status=True)
+    log.debug("process err %d" , err)
+    # check bucket shards num
+    bucket_stats1 = get_bucket_stats(ctx, client, bucket_name1)
+    assert  bucket_stats1.num_shards == num_shards
+    new_user_stats = get_user_stats(ctx, client, user)
+    assert old_user_stats == new_user_stats
+
+
+    # TESTCASE 'reshard-add','reshard','add','add non empty bucket to resharding queue','succeeds'
+    log.debug(' test: reshard add non empty bucket')
+    # create objs
+    num_objs = 8
+    for i in range(0, num_objs):
+        key = boto.s3.key.Key(bucket1)
+        key.key = 'obj' + `i`
+        key.set_contents_from_string(key.key)
+
+    old_user_stats = get_user_stats(ctx, client, user)
+    num_shards = bucket_stats1.num_shards + 1
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'add', '--bucket', bucket_name1, '--num-shards', str(num_shards)], check_status=True)
+
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'list'], check_status=True)
+    assert len(out) == 1
+    log.debug('bucket name %s', out[0]['bucket_name'])
+    assert out[0]['bucket_name'] == bucket_name1
+
+    # TESTCASE 'reshard process ,'reshard','process','reshard non empty bucket','succeeds'
+    log.debug(' test: reshard process non empty bucket')
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'process', '--bucket', bucket_name1], check_status=True)
+    # check bucket shards num
+    bucket_stats1 = get_bucket_stats(ctx, client, bucket_name1)
+    assert  bucket_stats1.num_shards == num_shards
+    new_user_stats = get_user_stats(ctx, client, user)
+    assert old_user_stats == new_user_stats
+
+    # TESTCASE 'dynamic reshard-process','reshard','','process bucket resharding','succeeds'
+    log.debug(' test: dynamic reshard bucket')
+    # create objs
+    bucket_stats2 = get_bucket_stats(ctx, client, bucket_name2)
+    old_num_shards = bucket_stats2.num_shards
+    num_objs = 11
+    for i in range(0, num_objs):
+        key = boto.s3.key.Key(bucket2)
+        key.key = 'obj' + `i`
+        key.set_contents_from_string(key.key)
+    old_user_stats = get_user_stats(ctx, client, user)
+
+    # check bucket shards num
+    (err, out) = rgwadmin(ctx, client, ['reshard', 'list'], check_status=True)
+    assert len(out) == 1
```

Comment:
a test run is showing a failure here:

http://qa-proxy.ceph.com/teuthology/cbodley-2019-05-23_20:37:05-rgw-wip-kefu-testing-2019-05-23-0009-distro-basic-smithi/3972928/teuthology.log

it may be due to:

> 2019-05-23T20:52:24.095 INFO:teuthology.orchestra.run.smithi084.stderr:parse error setting 'rgw_reshard_thread_interval' to '30' (Value '30' is below minimum 600)